packages/@lwc/engine-core/src/framework/hydration.ts

```diff
@@ -63,6 +63,10 @@ import type {
 import type { VM } from './vm';
 import type { RendererAPI } from './renderer';
 
+type Classes = Omit<Set<string>, 'add'>;
+
+const EMPTY_SET: Classes = new Set<string>();
```

Comment:
```suggestion
// Used as a perf optimization to avoid creating and discarding sets unnecessarily.
const EMPTY_SET: Classes = new Set<string>();
```